Cabal/Distribution/Backpack.hs

```diff
@@ -0,0 +1,57 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE PatternGuards #-}
+{-# LANGUAGE DeriveGeneric #-}
+module Distribution.Backpack (
```

Comment:
For the sake of people reading hereafter, perhaps most of these new modules could include a link to the general backpack readme/proposal so we can say "don't know what definite/hold/etc means? go read the overview here ..."

Indeed it's probably a good idea to keep a copy of the backpack spec in the Cabal repo (yes we ought to keep a live Cabal spec in the repo too).